### ISSUE TITLE
Is SCT compatible with the new Apple M1 chip?

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#3211]
The purpose of this issue is to document if the new Apple M1 chip is compatible with SCT.

Users having this new chip are welcome to report the experience, notably by running:
- Installation: `install_sct`
- Functional tests: `sct_testing`
- Unit tests: 
  ```bash
  source $SCT_DIR/python/etc/profile.d/conda.sh 
  conda activate venv_sct
  pytest $SCT_DIR/unit_testing
  ```
- Integration tests: `$SCT_DIR/batch_processing.sh`

Also see: https://github.com/shimming-toolbox/shimming-toolbox/issues/187

**Comment by mathieuboudreau** [comment]
Computer: MBP 13 inch M1 2020
OS: macOS Big Sur 11.2
CPU: Apple M1
RAM: 16 Gb

Here's are logs of my experience installing & running the tests in a normal terminal (not launched with Rosetta2 or using the `arch -x86_64` command):

## Installation

Followed the "Install from Github" instructions ([here](https://spinalcordtoolbox.com/en/stable/user_section/installation.html#install-from-github-development))

See log [here](https://docs.google.com/document/d/1eoBL-9UDzXZC6KsWG2nM10kfhG75HUh7SCSqGOM2dPk/edit?usp=sharing).

I didn't encounter or notice any errors.

## Functional tests

See log [here](https://docs.google.com/document/d/1QKPxXPPY6moM-yG_x8fogozStUgoKSTAmjOTBIrCF0I/edit?usp=sharing)

I didn't encounter or notice any errors.

</details>

## Unit tests

See log [here](https://docs.google.com/document/d/12nujdq8kklHPGAiWNY5xxupLW7eZPH1tZ7RpqvWByiQ/edit?usp=sharing)

I didn't encounter or notice any errors

## Integration tests

See log [here](https://docs.google.com/document/d/1oJa53NY8Q8MLC_n0xH8eMxMYTmKWieVtf_Tq6peXOlU/edit?usp=sharing).

GDrive folder qith QC output files can be found [here](https://drive.google.com/drive/folders/1wIC119ja24BU3Y9ldTwA6jssVaXfl2Yz?usp=sharing)

I didn't encounter or notice any errors.

🤷

**Comment by mathieuboudreau** [comment]
As far as I could tell, it wasn't using Rosetta2, however it could have possibly switch it on behind the scenes for some stuff without me knowing. I would of have to looked carefully at the Activity Monitor to be aware of it, it wouldn't of alerted me.

**Comment by mathieuboudreau** [comment]
I did another fresh installation & ran the tests and this time **two unit tests failed**, both in unit_testing/test_deepseg.py I think.  I might have run the wrong command the first time, as something didn't make sense to me.

[Here's the log.](https://docs.google.com/document/d/12iyYMuYAyE8RNLOIzP8j_dyHLa8VwJT2DXZgT150yAY/edit?usp=sharing)

Trying to run the tests with the prefix `arch -x86_64` made things worst, leading to several more tests failing, and hanging on the last test. [Log](https://docs.google.com/document/d/1WGOU-FU6Oy5rHrs2cuNQWkvsViNky8DtM87ay6Y4kuU/edit?usp=sharing)

**Comment by joshuacwnewton** [comment]
I'm fairly certain we can close this as resolved? Especially after the further investigations we took to resolve https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3566.

Additionally, while there have been issues on the forum that describe installation failures on M1 Macs (see https://forum.spinalcordmri.org/search?q=M1%20mac), all have been resolved. So, I think we're good at this point.
